src/EventListener/KernelRequestListener.php

```diff
@@ -0,0 +1,336 @@
+
+                                // Get current href attribute
+                                preg_match($atagRegex, $buffer, $matches);
+
+                                // Overwrite href attribute
+                                $buffer = preg_replace($atagRegex, 'id="splashImage_$1href="' . $strBlockUrl . urlencode($matches[2]) . '"', $buffer);
+                                $buffer = str_replace('iframe.src', 'iframe.setAttribute("data-ccb-id", "' . $cookie['id'] . '"); iframe.src', $buffer);
+
+                            } else
+                            {
+                                // Regex: Modify src attribute for iframes
+                                $frameRegex = "/<iframe([\s\S]*?)src=([\\\\\"\']+)(.*?)[\\\\\"\']+/i";
+
+                                // Get current src attribute
+                                preg_match_all($frameRegex, $buffer, $matches);
+
+                                $matchCount = count($matches[0]);
+                                for ($i = 0; $i < $matchCount; $i++)
+                                {
+
+                                    $quote = $matches[2][$i];
+                                    $search = 'src=' . $quote;
+                                    $replace = 'data-ccb-id=' . $quote . $cookie['id'] . $quote . '  src=' . $quote . $strBlockUrl . urlencode($matches[3][$i]) . $quote . ' data-src=' . $quote;
+
+                                    $iframe = str_replace($search, $replace, $matches[0][$i]);
+                                    $buffer = str_replace($matches[0][$i], $iframe, $buffer);
+
+                                }
+
+                            }
+
+                        }
+
+                    }
+
+                    break;
+
+                }
+
+            }
+
+        }
+
+        return $buffer;
+
+    }
+
+    /**
+     * @param ResponseEvent $event
+     * @return bool
+     * see also Contao\CoreBundle\EventListener\PreviewToolbarListener
+     */
+    private function isPageTemplate(ResponseEvent $event): bool
+    {
+        $request = $event->getRequest();
+        $response = $event->getResponse();
+
+        if (
+            $request->isXmlHttpRequest()
+            || (!$response->isSuccessful() && !$response->isClientError())
+        )
+        {
+            return false;
+        }
+
+        if (
+            'html' !== $request->getRequestFormat()
+            || !str_contains((string)$response->headers->get('Content-Type'), 'text/html')
+            || false !== stripos((string)$response->headers->get('Content-Disposition'), 'attachment;')
+        )
+        {
+            return false;
+        }
+
+        if (false === strripos($response->getContent(), '</body>'))
+        {
+            return false;
+        }
+
+        // @TODO Maybe this can be removed
+        if (!$this->scopeMatcher->isFrontendMainRequest($event))
+        {
+            return false;
+        }
```

Comment:
This currently does nothing. Either you will have to move it to the top, if you only want to apply this on the main request - or remove it entirely, if it should apply to fragment sub requests.

Reply:
What do you think? Is it ok to remove it or better palce as first condition?

Reply:
I don't know yet, I haven't reviewed the actual logic of your changes yet.